### PR TITLE
Use plist to manage arguments and environment for macos swarm plugin

### LINKED ISSUE #892: File jenkins-slave created with wrong owner under Darwin.
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 4.10.12
- Ruby: 2.3.7
- Distribution: Darwin (High Sierra)
- Module version:  1.7

## How to reproduce (e.g Puppet code you use)
`defaults.yaml`
```hiera
jenkins::slave::slave_user: build
jenkins::slave::slave_home: '/Users/build/.jenkins'
```

`site.pp`
```puppet
node default {
  include ::jenkins::slave
}
```

## What are you seeing

While deploying the catalog, some files are not respecting the fact that I'm overriding the Jenkins user. In my case it's coming from [slave](https://github.com/voxpupuli/puppet-jenkins/blob/master/manifests/slave.pp#L312-L313) lines 312-313.

This causes error as the user `jenkins` does not exist.

## What behaviour did you expect instead

I'd expect the files to get created with the correct user/group ownership. 

## Output log

```
Could not set 'file' on ensure: Could not find user jenkins at /etc/puppetlabs/code/thirdparty/jenkins/manifests/slave.pp:311 Wrapped exception: Could not find user jenkins
```

### PR DESCRIPTION
**Pull request by anutting**
#### Pull Request (PR) description
This change removes the start up script for macOS and moves the arguments to the plist template.

### CODE REVIEW

**Comment by anutting** [comment]
@rnelson0 ui_user is the user the swarm plugin uses to authenticate with the jenkins master (along with ui_pass). The slave class uses the puppet stdlib shellquoted function with ui_user then saves that to quoted_ui_user. The slave_user is the user account created on the slave to manage the swarm plugin and own any files.

File: templates/org.jenkins-ci.slave.jnlp.plist.erb [hunk 1]
```diff
@@ -2,11 +2,77 @@
 <!DOCTYPE plist PUBLIC "-/Apple/DTD PLIST 1.0/EN" "http:/www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+  <% if @quoted_ui_pass -%>
```
[the remark this answers is not in the document]
Comment (anutting) [reply templates/org.jenkins-ci.slave.jnlp.plist.erb]: sure, I'll get that going.

**Comment by rnelson0** [comment]
@anutting in #892, the issue is at lines 309-316 of `manifests/slave.pp` (and a few other places @dhoppe identified), but this PR does not modify those lines, which is why I'm not sure how it addresses the described failures.

**Comment by mmarseglia** [comment]
@rnelson0 the problem we found is that on macos the script used to invoke the service did not work.  we found that by removing the script and defining all variables in the plist the service started correctly.

**Comment by anutting** [comment]
@rnelson0 the file created on line 309 of manifests/slave.pp (jenkins-slave) used to be sourced from the start.slave.sh file created in manifests/slave.pp line 259 (which I removed in my commit). Since my PR no longer uses the jenkins-slave file, it fixes #892.

**Comment by Dan33l** [comment]
If you rebase you will get more dedicated messages during acceptance about fail due to error or idempotency. cf #900

**Comment by anutting** [comment]
@Dan33l Thank you for your suggestion to rebase master. It seems that didn't fix any of the Travis CI jobs.

@rnelson0 I've implemented your requested changes. Is there anything else needed to get this merged?

Thanks

**Comment by mmarseglia** [comment]
Given this might be a puppet6 issue and not with our PR may this be merged?

File: templates/org.jenkins-ci.slave.jnlp.plist.epp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  	<string><%= $jenkins::slave::quoted_ui_pass -%></string>
+  </dict>
+  <% } -%>
+  <key>Label</key>
+  <string>org.jenkins-ci.slave.jnlp</string>
+  <key>ProgramArguments</key>
+  <array>
+    <string>/usr/bin/java</string>
+    <% if $jenkins::slave::_real_java_args { -%>
+    <string><%= jenkins::slave::_real_java_args -%></string>
```
Comment (ekohl) [inline templates/org.jenkins-ci.slave.jnlp.plist.epp:18]: Missing a `$` here. Applied in https://github.com/voxpupuli/puppet-jenkins/pull/1057

**Review by ekohl — CHANGES_REQUESTED** [review]

**Comment by ekohl** [comment]
I've rebased this and fixed it up in https://github.com/voxpupuli/puppet-jenkins/pull/1057. I did notice labels were missing, but I see you already removed your fork so I think you have no more interest in fixing it up.

**Pull request closed without merging**